File: test-files/golden-tests/extensions/lua-register-transform/addons/extensions/transforms.lua
-- Declare two corpus transforms with `mrdocs.register_transform`. Both run, in
-- registration order, so one extension can contribute several
-- transforms. The first renames every function; the second rewrites
-- its brief.

mrdocs.register_transform(function(ctx)
    for _, sym in ipairs(ctx.corpus.symbols) do
        if sym.kind == "function" then
            sym.name = "renamed_" .. sym.name
        end
    end
end)

mrdocs.register_transform(function(ctx)
    for _, sym in ipairs(ctx.corpus.symbols) do
        if sym.kind == "function" then
            sym.doc = {
                brief = {
                    children = {
                        { kind = "text",
                          literal = "Brief from the second transform" }
                    }
                }
            }
        end
    end
end)

File: tests/golden/fixtures/extensions/js-register-transform/addons/extensions/transforms.js
// Declare two corpus transforms with `mrdocs.register_transform`. Both run, in
// registration order, so one extension can contribute several
// transforms. The first renames every function; the second rewrites
// its brief. Mirrors the lua-register-transform fixture on the JS path.

mrdocs.register_transform("rename-functions", function(ctx)
{
    for (var i = 0; i < ctx.corpus.symbols.length; ++i)
    {
        var sym = ctx.corpus.symbols[i];
        if (sym.kind === "function")
        {
            sym.name = "renamed_" + sym.name;
        }
    }
});

mrdocs.register_transform("set-brief", function(ctx)
{
    for (var i = 0; i < ctx.corpus.symbols.length; ++i)
    {
        var sym = ctx.corpus.symbols[i];
        if (sym.kind === "function")
        {
            sym.doc = {
                brief: {
                    children: [
                        { kind: "text",
                          literal: "Brief from the second transform" }
                    ]
                }
            };
        }
    }
});
